nimbusapp

```diff
@@ -75,95 +153,170 @@ fi
-  VERSION=$(grep ${REPOSITORY}/${IMAGE}.dockerapp ${ENV_FILE} | cut -d':' -f2 )
-  echo "Using version in ${ENV_FILE}: ${VERSION}"
+elif [[ -e $ENV_FILE ]]; then
+  product_exists
+  prodExists=$?
+  if [[ $prodExists -eq 0 ]]; then
+    #ensuring uniform comparision since someone could pass intillij or intellij.dockerapp
+    IMAGE=$(echo $IMAGE | cut -d'.' -f1 )".dockerapp"
+    VERSION=$(grep "^${IMAGE}" ${ENV_FILE} | cut -d':' -f2 )
+    output "Using ${IMAGE}:${VERSION} version found in ${ENV_FILE}"
+  else
+    output "Not able to find ${IMAGE} in ${ENV_FILE}.  Will attempt to start using ${IMAGE}:${VERSION}"
+  fi
 fi
 
 # IMAGE="admpresales/${IMAGE}.dockerapp"
 
 while [[ "$1" != "" ]]; do
-  PARAM=$1
-  VALUE=$2
-  case $PARAM in
-    -h | --help)
+  PARAM="$1"
+  VALUE="$2"
+
+  case "$PARAM" in
+    -h | --help | help)
       usage
       exit 0
       ;;
-      -s)
-         ARGS="${ARGS} $1 $2"
-         shift
-         shift
+      -d|--debug)
+        (( NIMBUS_DEBUG++ ))
+        if (( NIMBUS_DEBUG >= 2 )); then
+            BASH_VERBOSE="-v"
+        fi
+        shift
+      ;;
+      -s | --set)
+         append_dockerapp_arg "$PARAM" "$VALUE"
+         shift 2
          ;;
       -v)
-         # check to see if intellij is being called.  since args can appear in differ order, checking all 3 places:
-         # product, image or if it is a command not yet processed
-         if [[ (`echo ${PRODUCT}|grep "intellij"| wc -l` -eq 1) || (`echo ${IMAGE}|grep "intellij"| wc -l` -eq 1) || (`echo "$@" |grep "intellij"| wc -l` -eq 1)]]
+         # check to see if intellij is being called.
+        if [[ "${IMAGE}" == *intellij* ]]
          then
-            VOLUME_MOUNT="-e '\$a\    - ${HOME}/.m2:/home/demo/.m2' -e '\$a\    - ${HOME}/${IDEA}:/home/demo/IdeaProjects'"
+            # the \ \ forces spaces which are needed for the sed statment below for proper docker-compose format
+            VOLUME_MOUNT_IDEA="\ \ \ \ - type: bind\n\ \ \ \ \ \ source: ${HOME}\/${IDEA}\n\ \ \ \ \ \ target: \/home\/demo/\/IdeaProjects"
             create_intellij_mount_points
          else
-            echo "The '-v' option is only available with Intellij at this time.  Ignoring."
+            output "The '-v' option is only available with Intellij at this time.  Ignoring."
+         fi
+         shift
+         ;;
+      -m)
+         # check to see if intellij is being called.
+         if [[ "${IMAGE}" == *"intellij"* ]]
+         then
+            # the \ \ forces spaces which are needed for the sed statment below for proper docker-compose format
+            VOLUME_MOUNT_M2="\ \ \ \ - type: bind\n\ \ \ \ \ \ source: ${HOME}\/\.m2\n\ \ \ \ \ \ target: \/home\/demo\/\.m2"
+            create_intellij_mount_points
+         else
+            output "The '-m' option is only available with Intellij at this time.  Ignoring."
          fi
          shift
          ;;
       -p)
-         PROJECT=$2
+         PROJECT="${VALUE}"
```

Comment:
Project name needs to evaluate out to just the app, not the app.dockerapp, which is what seems to be happening right now. This will prevent someone who had been running from 1.0.0 from being able to use 1.1.0 to control any existing projects.

Output from verbose:
Running: docker-compose -p **octane.dockerapp** -f /home/demo/.nimbusapp/cache/octane.dockerapp/admpresales/octane.dockerapp/12.60.35.186/octane.dockerapp.yml up -d

Reply:
Addressed in 5b787f2 - default project name was not in line with new logic for figuring the image name

Reply:
LGTM